File: sorting/c/bubble_sort.c
#include "utils.h"
#include <stdio.h>

void swap(int *a, int *b) {
  int temp = *a;
  *a = *b;
  *b = temp;
}

void bubble_sort(int arr[], int n) {
  for (int i = 0; i < n - 1; ++i) {
    for (int j = 0; j < n - i - 1; ++j) {
      if (arr[j] > arr[j + 1]) {
        swap(&arr[j], &arr[j + 1]);
      }
    }
  }
}

int main() {
  int arr[] = {98, 2, -10, 9, 0, 7};
  int n = sizeof(arr) / sizeof(arr[0]);

  printf("Original array: ");
  printArray(arr, n);
  bubble_sort(arr, n);

  printf("Bubble sorted array: ");
  printArray(arr, n);
  return 0;
}

File: sorting/lua/bubble_sort.lua
-- Function to perform Bubble Sort on a table
function bubbleSort(arr)
  local n = #arr
  local swapped

  repeat
    swapped = false
    for i = 1, n - 1 do
      if arr[i] > arr[i + 1] then
        -- Swap elements if they are in the wrong order
        arr[i], arr[i + 1] = arr[i+1], arr[i]
        swapped = true
      end
    end
  until not swapped;
end

local numbers = {5, 2, 9, 1, 5, 6}
print("Original array:", table.concat(numbers, ", "))
bubbleSort(numbers)
print("Sorted array:", table.concat(numbers, ", "))
